hr_timesheet_overtime_begin_end/models/account_analytic_line.py

```diff
@@ -0,0 +1,28 @@
+# SPDX-FileCopyrightText: 2024 Coop IT Easy SC
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+
+from odoo import api, models
+
+
+class AnalyticLine(models.Model):
+    _inherit = "account.analytic.line"
+
+    @api.model
+    def _add_unit_amount_to_vals_list(self, vals_list):
+        # This is a kind of strange method. Like in hr_timesheet_overtime's
+        # create method, the value of unit_amount passed to create is _not_ the
+        # value written to the new record. Instead, the rate is applied on top
+        # of the provided value. This seems a little unusual/implicit/magical to
+        # me.
+        super()._add_unit_amount_to_vals_list(vals_list)
+        for vals in vals_list:
+            if vals.get("unit_amount") and (date := vals.get("date")):
+                vals["unit_amount"] *= self.rate_for_date(date)
+        return
```

Comment:
if the default value is correctly handled as explained in [my comment](https://github.com/OCA/timesheet/pull/692#discussion_r1650579647) on OCA/timesheet#692, this function shouldn’t be needed anymore.